src/main/kotlin/com/openlattice/mechanic/upgrades/DeleteOrgMetadataEntitySets.kt

```diff
@@ -0,0 +1,268 @@
+            }
+
+            auditEdgeEntitySetIds.forEach { deleteEntitySet(orgId, it) }
+            auditRecordEntitySetIds.forEach { deleteEntitySet(orgId, it) }
+
+            if (deleteEntitySet(orgId, entitySetId)) {
+                return deleteMetadataFromJson(orgId, metadataKey)
+            }
+            // if the esid is all zeros AND there's no audit entity sets, it's likely the org wasn't initialized
+            // correctly. since there's no entity sets to delete, we'll try deleting metadata from json.
+            else if (
+                entitySetId.toString() == "00000000-0000-0000-0000-000000000000"
+                && auditEdgeEntitySetIds.isEmpty()
+                && auditRecordEntitySetIds.isEmpty()
+            ) {
+                return deleteMetadataFromJson(orgId, metadataKey)
+            }
+        }
+        catch (e: Exception) {
+            logger.error("error deleting metadata and audit entity sets - org $orgId entity set $entitySetId", e)
+        }
+
+        return false
+    }
+
+    private fun deleteEntitySet(orgId: UUID, entitySetId: UUID): Boolean {
+
+        try {
+            logger.info("starting to delete entity set - org $orgId entity set $entitySetId")
+            val timer = Stopwatch.createStarted()
+
+            if (AuthorizingComponent.internalIds.contains(entitySetId)) {
+                throw ForbiddenException("$entitySetId is a reserved id and cannot be deleted")
+            }
+
+            val entitySet = toolbox.entitySets[entitySetId]
+            checkNotNull(entitySet) { "entity set is null - org $orgId entity set $entitySetId" }
+
+            val t1 = Stopwatch.createStarted()
+            val deleteJobId = dataDeletionService.clearOrDeleteEntitySet(entitySet.id, DeleteType.Hard)
+            waitUntilJobFinishedOrCanceled(deleteJobId, orgId, entitySet.id)
+            logger.info(
+                "dataDeletionService.clearOrDeleteEntitySet took {} ms - org $orgId entity set $entitySetId",
+                t1.elapsed(TimeUnit.MILLISECONDS),
+            )
+
+            val t2 = Stopwatch.createStarted()
+            entitySetService.deleteEntitySet(entitySet)
+            logger.info(
+                "entitySetService.deleteEntitySet took {} ms - org $orgId entity set $entitySetId",
+                t2.elapsed(TimeUnit.MILLISECONDS),
+            )
+
+            logger.info(
+                "deleting entity set took {} ms - org $orgId entity set $entitySetId",
+                timer.elapsed(TimeUnit.MILLISECONDS),
+            )
+            logger.info("finished deleting entity set - org $orgId entity set $entitySetId")
+            return true
+        } catch (e: Exception) {
+            logger.error("error deleting entity set - org $orgId entity set $entitySetId", e)
+        }
+
+        return false
+    }
+
+    private fun waitUntilJobFinishedOrCanceled(jobId: UUID, orgId: UUID, entitySetId: UUID) {
+
```

Comment:
Thread blocks until job status is one of { finished, paused, canceled } - the function name is confusing.

Reply:
Fixed.